### ISSUE TITLE
Convert all publiclab repositories' automated tests/checks from TravisCI to GitHub Actions

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#9641]
Multi-repository coordinating issue:

starting in https://github.com/publiclab/plots2/issues/8793, we developed a plan to move all PL repositories from Travis CI to GitHub Actions, and have made a lot of progress. But there are many repositories still to go, so we are coordinating here.

If you'd like to try, unfortunately the PR must be created by a maintainer (for security purposes) but we'd love help as far as it's possible.

Try looking in the "DONE" section for examples of successful conversions of projects with similar architecture!

### Remaining Travis => GitHub Actions migrations

- [ ] https://github.com/publiclab/Leaflet.DistortableImage 
- [ ] https://github.com/publiclab/infragram 
- [ ] https://github.com/publiclab/spectral-workbench 
- [ ] https://github.com/publiclab/mapknitter (https://github.com/publiclab/mapknitter/issues/1489)
- [ ] https://github.com/publiclab/webjack 
- [ ] https://github.com/publiclab/inline-markdown-editor 
- [ ] https://github.com/publiclab/leaflet-blurred-location 
- [ ] https://github.com/publiclab/leaflet-blurred-location-display 
- [ ] https://github.com/publiclab/mapknitter-exporter 
- [ ] https://github.com/publiclab/mapknitter-exporter-sinatra
- [ ] https://github.com/publiclab/webjack-firmata
- [ ] https://github.com/publiclab/image-sequencer-app 

### Done

- [x] https://github.com/publiclab/spectral-workbench.js (https://github.com/publiclab/spectral-workbench.js/pull/207)
- [x] https://github.com/publiclab/PublicLab.Editor (https://github.com/publiclab/PublicLab.Editor/pull/680)
- [x] https://github.com/publiclab/plots2 (https://github.com/publiclab/plots2/issues/8793)
- [x] https://github.com/publiclab/image-sequencer (https://github.com/publiclab/image-sequencer/pull/1783)
- [x] https://github.com/publiclab/leaflet-environmental-layers (https://github.com/publiclab/leaflet-environmental-layers/pull/503)
- [x] https://github.com/publiclab/community-toolbox (https://github.com/publiclab/community-toolbox/pull/435)

**Comment by govindgoel** [comment]
@jywarren I would start working on this.

**Comment by muokicaleb** [comment]
@jywarren I'd like to help with this issue.

**Comment by bajajvinamr** [comment]
@jywarren  I'd like to work on this

**Comment by jywarren** [comment]
Hi all - thank you for your support! I'm currently working on this in https://github.com/publiclab/spectral-workbench/pull/647 (my top priority) and in https://github.com/publiclab/mapknitter/issues/1489 (2nd priority). I've made progress on the first but slowly trying to eliminate remaining errors, reading the log carefully each time. Any input/help/tips/support appreciated!!!

**Comment by aliciapaz** [comment]
Hello @jywarren, @govindgoel, and everyone else! 
Today I'm officially joining the GitHub Actions Migrations Task Force (or GAMTF) :octocat:  🚣🏽‍♀️  (yes I invented the name myself)

Given the list above, what do you think is the best one to start?
My first thought is to go with the smaller repositories first (i.e: webjack?), but I'm open to suggestions.

Thank you all and have a wonderful week.

**Comment by jywarren** [comment]
Hi Alicia!!!!! Welcome!!! I'm so sorry I missed this as my notifications are quite narrowly tuned (you could imagine how many i get across all PL projects) - my sincere apologies. I've reached out on Gitter as well.

I think trying a smaller repository first to get the hang of it is a great idea. Then, taking them in order perhaps - i've ordered them with top = highest priority. And, I'll add your name to my notification filters! Thanks!!! May you have a wonderful week as well!

**Comment by jywarren** [comment]
This list is looking great! Excellent work, @aliciapaz!!

I'm also noting we will have to go back and add correct "CI passing" badges which I noticed in https://github.com/publiclab/image-sequencer#image-sequencer (still has the old Travis badge). 

The new badge is: https://github.com/publiclab/plots2/workflows/tests/badge.svg?branch=main

And we can link it to: https://github.com/publiclab/plots2/actions (or another more specific one like https://github.com/publiclab/plots2/actions/workflows/tests.yml?query=branch%3Amain ? I can't find a route showing just those tests run on the publiclab/plots2 main branch though...

**Comment by govindgoel** [comment]
@jywarren along with the CI changes we can change the badge too. Thanks for the new badge URL. Meanwhile, I will look wherever the changes are needed.

@aliciapaz I also missed this, but I liked the name you invented for the task. Thanks

**Comment by aliciapaz** [comment]
@govindgoel so you will change the badges in the already migrated repositories, am I right? 
@jywarren  I'm currently working on mapknitter (based on work by @cesswairimu, thanks!)
Having some issues but so far nothing different from the ones you had in SWB.
Hopefully, I'll get it done by today/tomorrow! 

Thanks to both!

**Comment by govindgoel** [comment]
@aliciapaz Yeah will work on that also will take up a repo to migrate to actions.

Thanks

**Comment by jywarren** [comment]
Super, thanks all!!!

**Comment by jywarren** [comment]
OK - i made a new checklist for the badges. If we could get a more specific link for the badge that would be great, though - only recent runs of "tests" on main branch of publiclab's fork?

**Comment by govindgoel** [comment]
Okay @jywarren will look for that.

**Comment by aliciapaz** [comment]
We can use GH's filters to achieve what @jywarren suggests:
For example: 
https://github.com/publiclab/plots2/actions/workflows/tests.yml?query=branch%3Amain
